pipeline/runtime/common.go

```diff
@@ -44,7 +44,7 @@ func getOutputVarCmd(entrypoint, outputVars []string, outputFile string) string
 		} else if isPython {
 			cmd += fmt.Sprintf("with open('%s', 'a') as out_file:\n\tout_file.write('%s ' + os.getenv('%s') + '\\n')\n", outputFile, o, o)
 		} else {
-			cmd += fmt.Sprintf(";echo \"%s $%s\" >> %s", o, o, outputFile)
+			cmd += fmt.Sprintf("\necho \"%s $%s\" >> %s", o, o, outputFile)
```

Comment:
Can you also understand why writing to file is required for step output variables when they are being read again in `fetchOutputVariables` method?

Reply:
Because in the le code we dont know the value of the exported output variables. Hence we print them to a file and then fetch them to save the values in the le code to be used in another step